src/custom-messages/controllers/custom-messages.controller.ts

```diff
@@ -2,40 +2,75 @@ import {
   Body,
   Controller,
   Delete,
+  ForbiddenException,
   Get,
   HttpCode,
   Param,
   Patch,
   Post,
-  Query,
+  Req,
 } from '@nestjs/common';
 import {
   ApiBadRequestResponse,
+  ApiForbiddenResponse,
+  ApiNotFoundResponse,
   ApiOperation,
   ApiResponse,
+  ApiUnauthorizedResponse,
 } from '@nestjs/swagger';
 
 import { CustomMessagesService } from '../services/custom-messages.service';
 
 import { CreateCustomMessageDTO } from '../entities/create-custom-messages.dto';
 import { CustomMessage } from '../entities/custom-messages.entity';
-import { GetCustomMessageDTO } from '../entities/get-custom-messages.dto';
 import { UpdateCustomMessageDTO } from '../entities/update-custom-messages.dto';
+import { User } from '../../users/entities/user.entity';
+import { Permission } from '../../auth/enums/permission.enum';
+import { subject } from '@casl/ability';
+import { CaslAbilityFactory } from '../../auth/casl/casl-ability.factory';
+import { Request } from 'express';
 
 @Controller('custom-messages')
 export class CustomMessagesController {
-  constructor(private readonly CustomMessagesService: CustomMessagesService) { }
+  constructor(
+    private readonly CustomMessagesService: CustomMessagesService,
+    private readonly caslAbilityFactory: CaslAbilityFactory,
+  ) {}
+
+  private async failIfCanNotAccess(
```

Comment:
pequeña sugerencia de ortografia que queda a tu gusto: si bien "can not" como dos palabras es valido en inglés, escribir "cannot" como una sola palabra es mucho mas comun por lo que yo pondria la n en minuscula